avocado/core/remote/runner.py

```diff
@@ -264,6 +265,13 @@ def run_suite(self, test_suite, variants, timeout=0, replay_map=None):
             self.result.end_tests()
             self.job._result_events_dispatcher.map_method('post_tests',
                                                           self.job)
+            # compare the major version of remote and local version of avocado. If different write to UI and job.log
+            if str(remote_version[0]) != version.MAJOR:
+                log = logging.getLogger("avocado.job")
+                self.job.log.info("WARNING    : avocado version on the remote system is %s, local is %s",
+                                  '.'.join(map(str, remote_version)), version.VERSION)
+                log.warn("WARNING    : avocado version on the remote system is %s, local is %s",
+                         '.'.join(map(str, remote_version)), version.VERSION)
```

Comment:
Last, something to think about: what is the best moment to show the message in the UI? After or before the tests? The way it is now, the message shows up after the tests, so you have to wait for the tests to finish to then see the message. Think in a long test (like `examples/tests/sleeptenmin.py`). Would it be good to have the information about version mismatch before the tests? I see some value in having the message in advance. In that case, the `APP_LOG.warn()` should be called before the line:
```python
            ...
            results = self.run_test(self.job.references, timeout)
            ...
```
On the other hand, as you know, the `TEST_LOG.warn()` have to wait for the `archive.uncompress(zip_path_filename, local_log_dir)`.

Your call. I have no strong opinion here.